### ISSUE TITLE
Support a MyMonero-style wallet

### ISSUE THREAD

**Issue by woodser** [monero-ecosystem/monero-javascript#4]
This issue requests support for an implementation of the MoneroWallet.js interface which allows the wallet's view key to be shared with a 3rd party in order to offload processing of incoming transactions.

By implementing the MoneroWallet.js interface, the adapter will conform to this library's specification and plug into existing tests.

**Comment by pitersky** [comment]
Hello! Our wallet software are now using MyMonero-compatible backend, we are considering to switch to monero-javascript implementation if we be able to run it alongside our backend services. @woodser what you think basically needs to be done on monero-javascript side to accomplish this? I am now lurking through the source, just looking how hard it may be to do it.

**Comment by woodser** [comment]
Ideally another implementation of [MoneroWallet.js](https://github.com/monero-ecosystem/monero-javascript/blob/master/src/main/js/wallet/MoneroWallet.js) is added to monero-javascript which supports a MyMonero-style backend.  Then applications like yours can use a single library and wallet interface to manage multiple wallet architectures. 
 Currently monero-javascript supports monero-wallet-rpc (MoneroWalletRpc.js) and fully local wallets using WebAssembly (MoneroWalletWasm.js and MoneroWalletKeys.js for local keys-only management) which can be used alongside your current implementation.

To support a MyMonero-style backend in monero-javascript, a new wallet implementation would be added, e.g. src/main/js/wallet/MoneroWalletRemoteViewKey.js, which extends and implements the methods of MoneroWallet.js.

For testing, we'd use the existing test suite to test the new wallet implementation by adding a new test class, e.g. src/test/TestMoneroWalletRemoteViewKey.js in the same style as the existing TestMoneroWalletRpc.js and other wallet test classes.

It would be great getting help to support such a backend by conforming what you have to the MoneroWallet.js interface for inclusion in monero-javascript.

**Comment by woodser** [comment]
@pitersky Do you have a link to your work for reference?

**Comment by pitersky** [comment]
Appreciate description very much, thank you, that helps a lot. I don't have any code to share yet, if you mean it by "my work". Anyway I have started to work on this issue. I plan to make a PR if I will get it to work.

**Comment by devinpearson** [comment]
@woodser would something like this speak to what you're looking to do here.

https://github.com/mymonero/mymonero-utils/tree/master/packages/mymonero-wallet-manager

This package uses https://github.com/mymonero/mymonero-utils/tree/master/packages/mymonero-lws-client for speaking to the server and https://github.com/mymonero/mymonero-utils/tree/master/packages/mymonero-monero-client to use the wasm code.

Trying to reduce the complexity of wasm files where possible is an ongoing process. The idea is to extend this to allow an rpc connection if a person chooses to use it that way.

**Comment by woodser** [comment]
@devinpearson Yes it looks like these assets could be used to implement the interface in monero-javascript, `MoneroWallet.js`, by sharing the wallet's view key with a light wallet server for scanning.

**Comment by trasherdk** [comment]
And this one [Monero light-wallet Server](https://github.com/vtnerd/monero-lws)

**Comment by devinpearson** [comment]
@trasherdk monero-lws is supported in the package I’ve linked. OpenMonero should be as well.
